lta/bundler.py

```diff
@@ -279,6 +311,10 @@ def __init__(self, config: Dict[str, str], logger: Logger) -> None:
     """Run bundler work cycles as an infinite loop."""
     bundler.logger.info("Starting work loop")
     while True:
+        # if there is a drain semaphore, don't do any additional work
+        if check_drain_semaphore():
+            bundler.logger.info("Component drained; shutting down.")
+            sys.exit(0)
```

Comment:
Same as above, putting it in the `while` condition.

Reply:
Yep.